src/pages/api/profile/get-address.ts

```diff
@@ -0,0 +1,115 @@
+import { getServerSideCookies, Endpoint, HttpStatus } from "@fwl/web";
+import {
+  loggingMiddleware,
+  wrapMiddlewares,
+  tracingMiddleware,
+  errorMiddleware,
+  recoveryMiddleware,
+  rateLimitingMiddleware,
+} from "@fwl/web/dist/middlewares";
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { Handler } from "@src/@types/handler";
+import { UserCookie } from "@src/@types/user-cookie";
+import { configVariables } from "@src/configs/config-variables";
+import { logger } from "@src/configs/logger";
+import { initProfileClient } from "@src/configs/profile-client";
+import getTracer from "@src/configs/tracer";
+import { ERRORS_DATA, webErrorFactory } from "@src/errors/web-errors";
+import { authMiddleware } from "@src/middlewares/auth-middleware";
+import { sentryMiddleware } from "@src/middlewares/sentry-middleware";
+import { getProfileAccessToken } from "@src/utils/get-profile-access-token";
+
+const handler: Handler = async (request, response) => {
+  const webErrors = {
+    invalidProfileToken: ERRORS_DATA.INVALID_PROFILE_TOKEN,
+    invalidScopes: ERRORS_DATA.INVALID_SCOPES,
+    profileDataNotFound: ERRORS_DATA.PROFILE_DATA_NOT_FOUND,
+    unreachable: ERRORS_DATA.UNREACHABLE,
+    invalidQueryString: ERRORS_DATA.INVALID_QUERY_STRING,
+    addressNotFound: ERRORS_DATA.ADDRESS_NOT_FOUND,
+  };
+
+  return getTracer().span("get-address handler", async (span) => {
+    const userCookie = await getServerSideCookies<UserCookie>(request, {
+      cookieName: "user-cookie",
+      isCookieSealed: true,
+      cookieSalt: configVariables.cookieSalt,
+    });
+
+    if (!userCookie) {
+      response.statusCode = HttpStatus.TEMPORARY_REDIRECT;
+      response.setHeader("location", "/");
+      response.end();
+      return;
+    }
+
+    const addressId = request.query.addressId;
```

Comment:
[2/3]
This should be a url param instead of a query param.
The way you wrote this file atm suggest that we would have a `api/profile/update-address.ts` at some point (and maybe a delete and get-addresses, etc...) but I really think we should handle those as REST endpoints:
```
GET /api/profile/addresses
GET /api/profile/addresses/:id
POST /api/profile/addresses
PATCH /api/profile/addresses/:id
DELETE /api/profile/addresses/:id
```

Reply:
I used to do it this way, but we agreed with fenn many months ago to make an API handler for each HTTP verbs, instead of making one which check the verb asked.

Reply:
oh 🤔
@Fenntasy can we have your opinion on this 🙏 ?

Reply:
Linking this comment as the context is not the same that a yeah ago https://github.com/fewlinesco/connect-account/pull/526#discussion_r623755429